packages/kbn-es-query/src/es_query/migrate_filter.ts

```diff
@@ -65,6 +65,13 @@ export function migrateFilter(filter: Filter, indexPattern?: DataViewBase) {
 
   if (!filter.query) {
     filter.query = {};
+  } else {
+    // handle the case where .query already exists and filter has other top level keys on there
+    Object.keys(filter).forEach((key) => {
```

Comment:
It seems to me that this code is not quite optimal in terms of performance. I see at least 3 iterations (`Object.keys`, `forEach`, `includes`) here, which seem to be not very necessary. What do you think about 3 simple if's

```
if ('meta' in filter) {
    delete filter['meta']
}
```

or if you still want to remove some duplication 

```
['meta', 'query', '$state'].forEach((key) => {
  if (key in filter) {
    delete filter[key]
  }
})


```